### DISCUSSION TITLE
How to use Pygame-ce?

### DISCUSSION THREAD

**Discussion by Mad-hu** [pyscript/pyscript#2227] [Q&A]
```
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>PyScript Example with Pygame-CE</title>
  <link rel="stylesheet" href="https://pyscript.net/snapshots/2024.9.2/core.css"/>
  <script type="module" src="https://pyscript.net/snapshots/2024.9.2/core.js"></script>
</head>
<body>
  <h1>PyScript Example with Pygame-CE</h1>

  <!-- 使用 <py-config> 标签指定需要安装的库 -->
  <py-config>
  {
      "packages": ["pygame-ce"]
  }
  </py-config>

  <!-- PyScript 标签，用于编写 Python 代码 -->
  <py-script>
    import asyncio

    async def main():
        import pygame

        # 初始化 Pygame
        pygame.init()
        screen = pygame.display.set_mode((640, 480))
        pygame.display.set_caption("Pygame-CE in PyScript")

        # 绘制一个简单的图形
        running = True
        while running:
            for event in pygame.event.get():
                if event.type == pygame.QUIT:
                    running = False

            screen.fill((255, 255, 255))
            pygame.draw.circle(screen, (0, 0, 255), (320, 240), 50)
            pygame.display.flip()

        pygame.quit()

    asyncio.ensure_future(main())
  </py-script>
</body>
</html>
```

![image](https://github.com/user-attachments/assets/fad8f397-9be7-4a20-8183-e3e91f5fcab3)

**Comment by giohappy** [comment]
From my understanding, we need to find a way to configure the [pyodide.canvas](https://pyodide.org/en/stable/usage/sdl.html#setting-canvas), but I haven't been able to access the `pyodide` JS interface inside `pyscript` yet.

I've found a commit where the runtime [was exposed globally](https://github.com/pyscript/pyscript/pull/868/files#diff-a46bdf2eb41054c052aba50dd29116d21635c57e69de9b2596015f9670ab7a8d) in `pyscript`, but I suppose today [polyscript](https://pyscript.github.io/polyscript/#bootstrapping-core) should be used to access the interpreter. 
I don't know... the docs don't help much and `pyscript` is getting more and more intricate.

**Comment by WebReflection** [comment]
there is a branch https://github.com/pyscript/pyscript/pull/2265 ... it's experimental but it works, maybe some code in there could help? We haven't landed that yet though ... should we? /cc @ntoll 

**edit** that branch can be tested out of `https://cdn.jsdelivr.net/npm/@pyscript/core@0.6.23/dist/core.js` as JS source and `https://cdn.jsdelivr.net/npm/@pyscript/core@0.6.23/dist/core.css` for CSS

**Reply by ntoll** [reply]
If you're happy with the code, I'm happy to merge with one caveat: let's discuss this at tomorrow's community call so it gets visibility in the wider community. I'll also mention this on discord. (I'm being ultra cautious here because adding a new script type is a relatively "big" change that would be hard to back out of.)

**Reply by WebReflection** [reply]
agreed on everything *but* it's **experimental** and we can add an *experimental* warning once enabled that will go once it's not experimental anymore. People pinning versions will be eventually happy or disappointed happily ever after, we don't bring in something we can't remove later on ... true that most experiments are still there, yet I think we did it before, we can keep rolling this way.

**Reply by ntoll** [reply]
Totally. We're completely aligned on this.
